server/src/main/java/io/crate/execution/dml/ObjectIndexer.java

```diff
@@ -59,17 +60,23 @@ public class ObjectIndexer implements ValueIndexer<Map<String, Object>> {
     private final HashMap<String, ValueIndexer<Object>> innerIndexers;
     private final ColumnIdent column;
     private final Function<ColumnIdent, Reference> getRef;
-    private final RelationName table;
+    private final DocTableInfo table;
     private final Reference ref;
     private final Function<ColumnIdent, FieldType> getFieldType;
     private final HashMap<String, DataType<?>> innerTypes;
+    private final Function<Reference, String> sourceKeyWriter;
 
     @SuppressWarnings("unchecked")
-    public ObjectIndexer(RelationName table,
+    public ObjectIndexer(DocTableInfo table,
                          Reference ref,
                          Function<ColumnIdent, FieldType> getFieldType,
                          Function<ColumnIdent, Reference> getRef) {
         this.table = table;
+        if (table.versionCreated().onOrAfter(Version.V_5_5_0)) {
+            sourceKeyWriter = reference -> Long.toString(reference.oid());
+        } else {
+            sourceKeyWriter = reference -> reference.column().leafName();
+        }
```

Comment:
This code / switch has already happen at the `Indexer`, so instead of passing the table and doing the logic again, we should pass in the `sourceKeyWriter`.

Reply:
Thanks for the suggestion, dropped second commit and did this instead.